### ISSUE TITLE
"Failed to create IDirectSound8 Object" when using headphones

### ISSUE THREAD

**Issue by Jer1605** [VFansss/mgs2-v-s-fix#56]
Hello !

First, thanks to you clever and well explained patch ! I post there an issue, that was already post before, but unresolved.

https://github.com/VFansss/mgs2-v-s-fix/issues/47

I have the same problem, and I hope you could help me if I send my log files. How can I send to you my mgs2.ini file ?
I want really bad to do the MGS serie again, you would be my savior !

[last.log](https://github.com/VFansss/mgs2-v-s-fix/files/2472052/last.log)
[dsoundlog.txt](https://github.com/VFansss/mgs2-v-s-fix/files/2472053/dsoundlog.txt)
[err.log](https://github.com/VFansss/mgs2-v-s-fix/files/2472054/err.log)

and if that can help ::

![sans titre](https://user-images.githubusercontent.com/22455472/46855147-2bced880-ce03-11e8-809a-71f43cf2b676.png)

Thanks very much !

**Comment by Jer1605** [comment]
here is the content of the mgs2.ini file 

`0003	0780
0004	0438
0070	4E564944
0071	49412047
0072	65466F72
0073	63652047
0074	54582031
0075	30363000
0005	0001
0006	0800
0007	0400
0031	0200
0032	0200
0040	0001
0010	0001
0048	0100
0049	0100
0047	0001
0041	0001
0042	0001
0043	0001
004A	0001
0044	0001
0045	0001
0090	5072696D
0091	61727920
0092	536F756E
0093	64204472
0094	69766572
009D	0002
009E	0009
009F	0002
0002	0001
0030	0004
0046	0001
`

**Comment by VFansss** [comment]
Hi! 

I didn't solved it yet because I don't have a DS4, so is pretty harsh to solve it without any old school "trial-and-error".

But if you like we could improvise a solution, and maybe I can get closer to the solution.

> How can I send to you my mgs2.ini file ?

You can attach files here simply clicking on the bar below the text area where you can reply to this comment.

> I have the same problem, and I hope you could help me if I send my log files.

Activate the [debug mode](https://github.com/VFansss/mgs2-v-s-fix/wiki/Troubleshooting-&-Debug-mode#debug-mode), configure your games with a working config,  then send me the mgs2.ini file and the files written on the debug mode wiki page!

**Comment by Jer1605** [comment]
Thank you for your quick answer !

here are the files (I had to change mgs2.ini format, because github doesn't accept "*.ini" extension, you remove the ".txt" extension I had, and you'll get the ini file normaly ;) ) ::

[MGS2_VFix_debug.txt](https://github.com/VFansss/mgs2-v-s-fix/files/2472360/MGS2_VFix_debug.txt)
[dsoundlog.txt](https://github.com/VFansss/mgs2-v-s-fix/files/2472366/dsoundlog.txt)
[err.log](https://github.com/VFansss/mgs2-v-s-fix/files/2472367/err.log)
[last.log](https://github.com/VFansss/mgs2-v-s-fix/files/2472369/last.log)
[mgs2.ini.txt](https://github.com/VFansss/mgs2-v-s-fix/files/2472373/mgs2.ini.txt)

**Comment by VFansss** [comment]
Are you using the headphones through a dualshock4 gamepad, am I right?

Please answer, because you didn't stated it explicitly, and this is truly important to pinpoint the real cause of the issues.

I've made a little software with an algorithm that auto-detect audio peripherals.

Please download it, extract the .zip and run the .exe

[AudioOutDetector.zip](https://github.com/VFansss/mgs2-v-s-fix/files/2472400/AudioOutDetector.zip)
 
It will print an output on a textblock at the center of the screen.

Please paste here its output!

**Comment by Jer1605** [comment]
No, I don't use any dualshock4 gamepad. 
I am on a laptop : PC Portable Asus ROG Strix-GL503VM-FY337T 15.6" Gaming

Here is the output of your program : 

> detected devices


Speakers (Realtek High Definiti

**Comment by VFansss** [comment]
> No, I don't use any dualshock4 gamepad.

Ok, thing change completely my view about the issue.

Let's do a test. Download this .zip and extract the mgs2.ini inside the game folder, and overwrite.

[Test1.zip](https://github.com/VFansss/mgs2-v-s-fix/files/2472549/Test1.zip)

then start the game from mgs2_sse.exe without opening the fix.

Tell me if if something changes...

**Comment by Jer1605** [comment]
Nope,

the error always pop : 

1. "Failed to create IDirectSound8 Object"
2. Abnormal shutdown

**Comment by VFansss** [comment]
Ok, there's some work to do.

I've built some similar test.

Please do them all, one by one, and tell me if something change.

[Test 2.zip](https://github.com/VFansss/mgs2-v-s-fix/files/2472621/Test.2.zip)

[Test 3.zip](https://github.com/VFansss/mgs2-v-s-fix/files/2472622/Test.3.zip)

[Test 4.zip](https://github.com/VFansss/mgs2-v-s-fix/files/2472623/Test.4.zip)

[Test 5.zip](https://github.com/VFansss/mgs2-v-s-fix/files/2472624/Test.5.zip)

**Comment by Jer1605** [comment]
Nothing changes.
I don't know if it changes anything, but i'm on windows 10, running the mgs2_sse.exe file on compatibility mode for windows XP (service pack 3)

**Comment by VFansss** [comment]
> I don't know if it changes anything, but i'm on windows 10, running the mgs2_sse.exe file on compatibility mode for windows XP (service pack 3)

No, that isn't important at all.

Just another shot: inside the game folder there is a file called "_MGS2SConfig.exe.oldandcrappy".

Rename it in "_MGS2SConfig.exe" and start it. It's the old game setupper.

Press OPTION -> SOUND -> Sound Driver, and tell me what voices appear in the drop-down menu.

**Comment by Jer1605** [comment]
![audio](https://user-images.githubusercontent.com/22455472/46864576-1bc2f300-ce1b-11e8-9533-532876c35a16.png)

**Comment by VFansss** [comment]
Try to select the second option and press OK than start the game.

**Comment by Jer1605** [comment]
It doesn't work either ! Strange ! Maybe I should buy a PS2 ;D

**Comment by VFansss** [comment]
> Maybe I should buy a PS2 ;D

Well, for sure wouldn't be a bad deal at all.

Honestly this bug is pretty sneaky 😞 

The worst things at all is that I didn't succeed in replicate it in any way. I've tried it personally on 4 different configurations, and I didn't never had any problem.

I can't totally understand if the cause is that the game try to output the sound to a wrong device, or is something out of place with an external library of some kind.

Try to lower quality of sound to 11 or 22 khz from the 'Audio tab', or try to disable the "NVidia Output" you have on output devices.

Also, the game is correctly working without headphones, right? (= if you use PC speakers instead of plug in the headphones into the pc jack )

Can you send me a screenshot of the windows output devices (the windows you sent me in the first message) **WITH** and **WITHOUT** the headphones plugged in?

**Comment by Jer1605** [comment]
Thanks for your time !
No, the game doesn't work either with or without headphones. I'll try to disable Nvidia Output !

**Comment by VFansss** [comment]
> Thanks for your time !

No, thanks to you and your patience. Unfortunately, this time I'm having big problem on solving this.

> No, the game doesn't work either with or without headphones.

This was a very important detail. The title of your issue is 

"Failed to create IDirectSound8 Object" **when** using headphones

With that title, I assumed that it worked good without headphones, but now I've discovered that I was wrong the whole time. It was a very important detail, and it made me focused on a wrong things.

You can try to delete "_audio_fix_applied.sss", "dsound.dll" and "dsound_x64.dll".

Also, you can try to lower the sampling rate doing this:

* Push ⊞ Win+R, then type mmsys.cpl.
* Find the Default playback device, marked by a green icon, and Double Click on it.
* Select the Advanced tab, then under the Default Format header, click on the drop down menu.
* Set the playback frequency to 44,100 Hz, either 16bit or 24bit, then hit OK.

**Comment by Jer1605** [comment]
I cant, I only have the choice between 
![image](https://user-images.githubusercontent.com/22455472/46872389-3bb2e080-ce34-11e8-9c86-42f448f2a730.png)

**Comment by VFansss** [comment]
Have you tried to delete "_audio_fix_applied.sss", "dsound.dll" and "dsound_x64.dll"?

**Comment by Jer1605** [comment]
Oh yes ! I can access to the main menu ! thank you very much for your help !!

**Comment by VFansss** [comment]
So the deletion of these files has worked? The sound is working good?

Beware that opening the fix will probably re-create from scratch these files, resulting again in the same problem.

Can you please do that, let the fix recreate all of them and THEN delete only "dsound.dll" anche check if its working?

I don't understand perfectly why of this situation, but I think I have to sort it myself.

**Comment by UltimateNova1203** [comment]
I would like to add to this, as I have the DS4 Wireless USB Adapter (Sony's official one) and would also like to get the audio to work when using the headphones. The USB adapter was referenced in another issue, but it appears to stem from the same core problem listed below.

 It does appear to be a sample rate issue, while I am limited in the rates I can test, I did try a few different ones and found some pattern. Bit-ness doesn't appear to matter, Hz does appear to matter, Unkown at this time if channels matter.

I have tried it with a few different devices (one of them is actually the same device), and have found that my devices function a so:

- Dualshock 4 USB Wireless Adaptor, format 2 channel, 16 bit, 32000Hz, fails
- Bluetooth Headset Hands-free, format 1 channel, 16 bit, 16000Hz, fails
- Same Bluetooth Headset Stereo, format 2 channel, 16 bit, 44100Hz, works
- Realtek Audio Speakers, format 16 bit, 48000Hz, works (Assuming 2 channel, doesn't report)
- Realtek Audio Speakers, format 24 bit, 48000Hz, works (Assuming 2 channel)

AudioTool reports the following:
detected devices
Headset (Q30 Hands-Free)
Headset Earphone (DUALSHOCK®4 U
Speakers / Headphones (Realtek
Headphones (Q30 Stereo)

If I remember correctly, in 2015 I was modifying MGS1 and MGS2 PC audio, and ran into sample rate issues when trying to get MGS1 to run with different audio (for funsies), it had a limit around 22000Hz. It's very likely MGS2 doesn't like anything lower than 44100Hz and cannot resample lower than this, I blame buggy code from porting.

I tried the fixes listed above, but would not run on the devices with lower sample rates. Game runs perfectly fine on sample rates 44100Hz and higher. Below is the audio settings for each device. I will have to wait to see if the Steam Remote Play allows other sample rates, but the app is very buggy for audio to Android, so may not be the best test.

![audiosupersheet](https://user-images.githubusercontent.com/19739806/50888977-104c7d00-13b4-11e9-8f26-218cfbcda0bb.png)
